### ISSUE TITLE
All plugin data suddenly reset

### ISSUE THREAD

**Issue by hypoflex** [GreepTheSheep/openplanet-maniaexchange-menu#25]
All my favorite saved maps are suddenly gone. entire list is empty.

OpenplanetNext\PluginStorage\ManiaExchange folder only has PlayLater.json
which is also empty. 

are they stored somewhere else? has something changed?

**Comment by GreepTheSheep** [comment]
Seems something else than the plugin deleted the folder. I don't destroy the PluginStorage of the plugin unless something got corrupted https://github.com/GreepTheSheep/openplanet-maniaexchange-menu/blob/14c7c101c0b6ae7fd006c7865d604b38c69b92ea/src/Utils/Methods.as#L37-L59

If you got a warning notification before, that should be the case.

> are they stored somewhere else?

No, I just save the Play Later playlist on a single file.

> has something changed?

No, since the version 1.2 of the plugin (last year), the file was moved from Openplanet user folder to PluginData (but I think that should not be the case for you, just for info)
https://github.com/GreepTheSheep/openplanet-maniaexchange-menu/blob/14c7c101c0b6ae7fd006c7865d604b38c69b92ea/src/Utils/Methods.as#L23

**Comment by hypoflex** [comment]
https://github.com/GreepTheSheep/openplanet-maniaexchange-menu/assets/36196198/7b23d68f-5726-4e57-82e5-f1be3836fb8f

Checked the log to see if something came up, it does report it adds it but nothing happens.
see attached video

**Comment by GreepTheSheep** [comment]
Favorites maps are linked with your Trackmania account, instead of the Play Later list (witch uses local data).

If it doesn't load, that should be an issue with the Trackmania API (load times as shown in the video)

You can try force reload by using Advanced>Refresh Favorites

**Comment by hypoflex** [comment]
Ahh okay, that makes the rabbit hole even more weird. As when I look on the exchange website (logged in with the same ubi account) I see one in my favorites. Which is also weird.

That one doesn't show up in game either.  I tried the force yesterday as well.  But let me try again and see if log shows something.

Else might have to add some additional logging to see what is going on.

**Comment by GreepTheSheep** [comment]
You can get more trace logging if you enable Developer Mode, you need to go to Developer>Signature mode>Developer. It'll reload Openplanet and enable the mode for this session only. It shows more logs on ManiaExchange

**Comment by hypoflex** [comment]
Sadly I'm also unaware of when the problem started. I only remember adding a few maps to the favorite last Saturday, which I wanted to add to my server yesterday when I noticed the list was empty.

I did install the BetterRoomManager plugin also Saturday, so will see if that may interfere. 

for now, this is the log in developer mode, attached full log below as well:

```[   ScriptRuntime] [18:15:09] [ManiaExchange]  MapTab::StartRequest (MX): https://trackmania.exchange/api/maps/get_map_info/multi/173979
[   ScriptRuntime] [18:15:09] [ManiaExchange]  MapTab::StartRequest (Authors): https://trackmania.exchange/api/maps/get_authors/173979
[   ScriptRuntime] [18:15:09] [ManiaExchange]  MapTab::CheckRequest (MX): [{"TrackID":173979,"UserID":61811,"Username":"Vekisis","GbxMapName":"$3F0T$5D3r$8A6i$A89a$D5Cl$C0Cs Of The $0C0Dr$2B0y$2A0ad","AuthorLogin":"5e1anP2vTIqHK1Dac5-RpA","MapType":"TM_Race","TitlePack":"TMStadium","TrackUID":"nnlK9Ml4R9QB5iCqJtF4C_ijDHd","Mood":"48x48Day","DisplayCost":18842,"ModName":"","Lightmap":8,"ExeVersion":"3.3.0","ExeBuild":"2024-04-30_16_52","AuthorTime":272485,"ParserVersion":2,"UploadedAt":"2024-05-31T19:26:51.367","UpdatedAt":"2024-05-31T19:39:05.613","Name":"Trials Of The Dryad","Tags":"24,33","TypeName":"Race","StyleName":null,"EnvironmentName":"Stadium","VehicleName":"CarSport","UnlimiterRequired":false,"RouteName":"Single","LengthName":"4 m 30 s","DifficultyName":"Advanced","Laps":1,"ReplayWRID":null,"ReplayWRTime":null,"ReplayWRUserID":null,"ReplayWRUsername":null,"TrackValue":0,"Comments":"","MappackID":0,"Unlisted":false,"Unreleased":false,"Downloadable":true,"RatingVoteCount":0,"RatingVoteAverage":0.0,"HasScreenshot":false,"HasThumbnail":true,"HasGhostBlocks":true,"EmbeddedObjectsCount":78,"EmbeddedItemsSize":4677402,"AuthorCount":1,"IsMP4":true,"SizeWarning":false,"AwardCount":17,"CommentCount":4,"ReplayCount":0,"ImageCount":0,"VideoCount":0}]
[   ScriptRuntime] [18:15:09] [ManiaExchange]  MapTab::CheckRequest (Authors): [{"UserID":61811,"Username":"Vekisis","Role":"","Uploader":true}]
[   ScriptRuntime] [18:15:09] [ManiaExchange]  NadeoServices - Check if map exists: https://live-services.trackmania.nadeo.live/api/token/map/nnlK9Ml4R9QB5iCqJtF4C_ijDHd
[   ScriptRuntime] [18:15:12] [ManiaExchange]  NadeoServices - Add map to favorites: https://live-services.trackmania.nadeo.live/api/token/map/favorite/nnlK9Ml4R9QB5iCqJtF4C_ijDHd/add
[   ScriptRuntime] [18:15:12] [ManiaExchange]  NadeoServices - Map favorite created: nnlK9Ml4R9QB5iCqJtF4C_ijDHd
[   ScriptRuntime] [18:15:12] [ManiaExchange]  NadeoServices - Loading Favorite tracks...
[   ScriptRuntime] [18:15:12] [ManiaExchange]  NadeoServices - Loading favorite maps: https://live-services.trackmania.nadeo.live/api/token/map/favorite?offset=0&length=100&sort=date&order=desc
[   ScriptRuntime] [18:15:12] [ManiaExchange]  NadeoServices - Check favorite maps: {"mapList":[],"itemCount":0}
```

[Openplanet.log](https://github.com/user-attachments/files/15536797/Openplanet.log)
